Framework/Algorithms/src/FilterEvents.cpp

```diff
@@ -736,6 +737,8 @@ void FilterEvents::splitTimeSeriesProperty(
       // add property to the associated workspace
       DataObjects::EventWorkspace_sptr ws_i = wsiter->second;
       ws_i->mutableRun().addProperty(output_vector[tindex], true);
+      Mantid::Geometry::Goniometer uniGonio = m_eventWS->run().getGoniometer();
```

Comment:
This is not the correct place to set the goniometer. You should add it next to this https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/src/FilterEvents.cpp#L277 And use `try...catch`. Don't forget to change `false` to `true`, so it recalculates the goniometer